packages/@lwc/ssr-compiler/src/compile-template/transformers/text.ts

```diff
@@ -21,19 +25,12 @@ import type { Transformer } from '../types';
 const bYield = (expr: EsExpression) => b.expressionStatement(b.yieldExpression(expr));
 
 const bYieldEscapedString = esTemplateWithYield`
-    const ${/* temp var */ is.identifier} = ${/* string value */ is.expression};
-    switch (typeof ${0}) {
-        case 'string':
-            yield (${/* is isolated text node? */ is.literal} && ${0} === '') ? '\\u200D' : htmlEscape(${0});
-            break;
-        case 'number':
-        case 'boolean':
-            yield String(${0});
-            break;
-        default:
-            yield ${0} ? htmlEscape(${0}.toString()) : ${2} ? '\\u200D' : '';
+    { 
+        const value = ${/* string value */ is.expression};
+        const massagedValue = value == null ? '' : String(value);
```

Comment:
Let's add a comment with a link to the code in `api.ts` to make it clear why we're doing this particular logic:

https://github.com/salesforce/lwc/blob/348130f1a03a6d90e350b504cd10602ed97a54fb/packages/%40lwc/engine-core/src/framework/api.ts#L548C18-L548C25

Reply:
@nolanlawson for the [suggestion about function return type](https://github.com/salesforce/lwc/pull/4963#discussion_r1859271559):  EsBlockStatement might be too specific as it can also [return an IRLiteral[] ](https://github.com/salesforce/lwc/blob/master/packages/%40lwc/ssr-compiler/src/compile-template/transformers/text.ts#L44)

Reply:
Ah, missed that. OK no problem.